frontend/fingertips-frontend/components/views/TwoOrMoreIndicatorsEnglandView/index.tsx

```diff
@@ -1,31 +1,25 @@
-import { TwoOrMoreIndicatorsEnglandDashboard } from '@/components/dashboards/TwoOrMoreIndicatorsEnglandDashboard';
+import { TwoOrMoreIndicatorsEnglandViewPlots } from '@/components/viewPlots/TwoOrMoreIndicatorsEnglandViewPlots';
 import { areaCodeForEngland } from '@/lib/chartHelpers/constants';
-import {
-  SearchParams,
-  SearchStateManager,
-  SearchStateParams,
-} from '@/lib/searchStateManager';
+import { SearchParams, SearchStateManager } from '@/lib/searchStateManager';
 import { connection } from 'next/server';
-
-type OneIndicatorTwoOrMoreAreasViewProps = {
-  searchState: SearchStateParams;
-};
+import { ViewProps } from '../ViewsContext';
 
 export default async function TwoOrMoreIndicatorsEnglandView({
   searchState,
-}: Readonly<OneIndicatorTwoOrMoreAreasViewProps>) {
+}: Readonly<ViewProps>) {
   const stateManager = SearchStateManager.initialise(searchState);
-  const {
-    [SearchParams.AreasSelected]: areaCodes,
-    [SearchParams.GroupSelected]: selectedGroupCode,
-  } = stateManager.getSearchState();
+  const { [SearchParams.AreasSelected]: areasSelected } =
+    stateManager.getSearchState();
 
-  const areasSelected = areaCodes ?? [];
+  if (
+    areasSelected! &&
```

Comment:
I don't think this needs the non-null assertion here:
```suggestion
    areasSelected &&
```

Reply:
I think I do in order to then make the next checks, but optional chainging seems better.

Reply:
Just having the `if(areasSelected ...` should be enough for Typescript to narrow the type of `areasSelected` so that it doesn't include undefined and you can then reference properties of it after that. I tried it quickly on my machine and it seemed happy enough with just removing the `!`. You're right though that optional chaining is neater.

In general, I'm always wary when I see the `!` operator being used. There are obviously some valid use cases, but it's too often used when not needed, or because some defensive coding that should probably be present hasn't been implemented.